File: extensions/instagram/src/main/java/app/revanced/extension/instagram/misc/download/DownloadMediaPatch.java
package app.revanced.extension.instagram.misc.download;

@SuppressWarnings("unused")
public class DownloadMediaPatch {

    /**
     * Injection point.
     * Always return true to allow downloading all media regardless of creator settings.
     */
    public static boolean isDownloadAllowed() {
        return true;
    }

    /**
     * Injection point.
     * Force third-party downloads to be enabled.
     */
    public static int forceThirdPartyDownloadsEnabled(int originalValue) {
        return 1;
    }
}

File: patches/src/main/kotlin/app/revanced/patches/instagram/misc/download/DownloadMediaPatch.kt
package app.revanced.patches.instagram.misc.download

import app.revanced.patcher.patch.bytecodePatch
import app.revanced.util.returnEarly

@Suppress("unused")
val downloadMediaPatch = bytecodePatch(
    name = "Download media",
    description = "Allows downloading all media (photos, videos, reels) regardless of creator's download settings.",
    use = false,
) {
    compatibleWith("com.instagram.android")

    apply {
        downloadAllowedMethodMatch.method.returnEarly()
    }
}
